control-plane/pkg/contract/subscriptionsapi.go

```diff
@@ -0,0 +1,196 @@
+	// key-value pair
+	for k, v := range f.Suffix {
+		p.Attribute = k
+		p.Suffix = v
+		// We only expect a single element, but let's break anyway
+		break
+	}
+
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Suffix{
+			Suffix: p,
+		},
+	}
+}
+
+// NewAllFilter converts the SubscriptionsAPIFilter into the all dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// All filter evaluates to true when all nested filters evaluate to true.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#all-filter-dialect
+func NewAllFilter(filters []v1.SubscriptionsAPIFilter) *DialectedFilter {
+	all := All{
+		Filters: []*DialectedFilter{},
+	}
+	for _, f := range filters {
+		all.Filters = append(all.Filters, FromSubscriptionFilter(f))
+	}
+	return &DialectedFilter{
+		Filter: &DialectedFilter_All{
+			All: &all,
+		},
+	}
+}
+
+// NewAnyFilter converts the SubscriptionsAPIFilter into the any dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// All filter evaluates to true when all nested filters evaluate to true.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#any-filter-dialect
+func NewAnyFilter(filters []v1.SubscriptionsAPIFilter) *DialectedFilter {
+	any := Any{
+		Filters: []*DialectedFilter{},
+	}
+	for _, f := range filters {
+		any.Filters = append(any.Filters, FromSubscriptionFilter(f))
+	}
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Any{
+			Any: &any,
+		},
+	}
+}
+
+// NewNotFilter converts the SubscriptionsAPIFilter into the not dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// Not filter evaluates to true when the nested filter evaluates to false.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#not-filter-dialect
+func NewNotFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Not{
+			Not: &Not{
+				Filter: FromSubscriptionFilter(f),
+			},
+		},
+	}
+}
+
+// NewCESQLFilter converts the SubscriptionsAPIFilter into the sql dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// CESOL filter is a Cloud Events SQL Expression
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/sql.md#any-filter-dialect
+//
+// See "CNCF CloudEvents SQL Expression Language"
+// https://github.com/cloudevents/spec/blob/main/cesql/spec.md
+func NewCESQLFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Cesql{
+			Cesql: &CESQL{
+				Expression: f.SQL,
```

Comment:
While `f.SQL` currently works (we have blockers on the dependency updates).

Soon this will be `CESQL` - but that's more a FYI (especially with a funky state for dependency updates, upstream :see_no_evil: )

https://github.com/knative/eventing/pull/6148/files#diff-9897c649cc6cb820cddfd787a041f960074f91d38b87bd24216e110e76718c1eR175

Reply:
Yep